arch/src/aarch64/mod.rs

```diff
@@ -195,22 +196,17 @@ fn get_fdt_addr() -> u64 {
 }
 
 pub fn get_host_cpu_phys_bits() -> u8 {
-    // The value returned here is used to determine the physical address space size
-    // for a VM (IPA size).
-    // In recent kernel versions, the maximum IPA size supported by the host can be
-    // known by querying cap KVM_CAP_ARM_VM_IPA_SIZE. And the IPA size for a
-    // guest can be configured smaller.
-    // But in Cloud-Hypervisor we simply use the maximum value for the VM.
-    // Reference https://lwn.net/Articles/766767/.
-    //
-    // The correct way to query KVM_CAP_ARM_VM_IPA_SIZE is via rust-vmm/kvm-ioctls,
-    // which wraps all IOCTL's and provides easy interface to user hypervisors.
-    // For now the cap hasn't been supported. A separate patch will be submitted to
-    // rust-vmm to add it.
-    // So a hardcoded value is used here as a temporary solution.
-    // It will be replace once rust-vmm/kvm-ioctls is ready.
-    //
-    40
+    // A dummy hypervisor created only for querying the host IPA size and will
+    // be freed after the query.
+    let hv = hypervisor::new().unwrap();
+    let host_cpu_phys_bits = hv.get_host_ipa_limit().try_into().unwrap();
```

Comment:
Why not convert simply with `let host_cpu_phys_bits = hv.get_host_ipa_limit() as u8;` ?

Reply:
Actually, in this content, we can, as the host ipa limit cannot go over than the limit of u8. The `try_into().unwrap()` will panic if the converted value wouldn't fit.

Reply:
Got it. Please keep it.